### PR TITLE
Add Knative as adopter

### PR DESCRIPTION
**Pull request by carlisia**
This PR adds Knative as a Contour adopter.

### CODE REVIEW

File: ADOPTERS.md [hunk 1]
```diff
@@ -16,7 +18,11 @@ _Add yours here!_
 
 Below is a list of solutions where Contour is being used as a component.
 
-**[VMware](https://tanzu.vmware.com/tanzu)**  
+**[Knative](https://knative.dev)**
+
+Knative can use Contour to serve all incoming traffic via the `net-contour` ingress Gateway. The net-contour controller enables Contour to satisfy the networking needs of Knative Serving by bridging Knative's KIngress resources to Contour's HTTPProxy resources.
```
Comment (stevesloka) [inline ADOPTERS.md:23]: nit: Maybe link out to the net-contour repo? (https://github.com/knative-sandbox/net-contour)

**Review by stevesloka — APPROVED** [review]
/lgtm with one small nit. 🎉

File: ADOPTERS.md (hunk 1 above)
Comment (carlisia) [reply ADOPTERS.md:23]: Yes, of course!! Thanks for thinking of this, changing..

**Review by skriss — APPROVED** [review]
LGTM, thanks @carlisia!

**Pull request merged**